dangerzone/gui/main_window.py

```diff
@@ -585,6 +631,96 @@ def dangerous_doc_button_clicked(self) -> None:
             pass
 
 
+class DocSelectionDropFrame(QtWidgets.QFrame):
+    """
+    HACK Docs selecting widget "drag-n-drop" border widget
+    The border frame doesn't show around the whole widget
+    unless there is another widget wrapping it
+    """
+
+    documents_selected = QtCore.Signal(list)
+
+    def __init__(
+        self, dangerzone: DangerzoneGui, docs_selection_widget: DocSelectionWidget
+    ) -> None:
+        super().__init__()
+
+        self.dangerzone = dangerzone
+        self.docs_selection_widget = docs_selection_widget
+
+        # Drag and drop functionality
+        self.setAcceptDrops(True)
+
+        self.document_image_text = QtWidgets.QLabel(
+            "Drag and drop\n documents here\n\n or"
+        )
+        self.document_image_text.setAlignment(QtCore.Qt.AlignCenter)
+        self.document_image = QtWidgets.QLabel()
+        self.document_image.setAlignment(QtCore.Qt.AlignCenter)
+        self.document_image.setPixmap(
+            load_svg_image("document.svg", width=20, height=24)
+        )
+
+        self.center_layout = QtWidgets.QVBoxLayout()
+        self.center_layout.addWidget(self.document_image)
+        self.center_layout.addWidget(self.document_image_text)
+        self.center_layout.addWidget(self.docs_selection_widget)
+
+        self.drop_layout = QtWidgets.QVBoxLayout()
+        self.drop_layout.addStretch()
+        self.drop_layout.addLayout(self.center_layout)
+        self.drop_layout.addStretch()
+
+        self.setLayout(self.drop_layout)
+
+    def dragEnterEvent(self, ev: QtGui.QDragEnterEvent) -> None:
+        ev.accept()
+
+    def dragLeaveEvent(self, ev: QtGui.QDragLeaveEvent) -> None:
+        ev.accept()
+
+    def dropEvent(self, ev: QtGui.QDropEvent) -> None:
+        ev.setDropAction(QtCore.Qt.CopyAction)
+        documents = []
+        for url_path in ev.mimeData().urls():
+            doc_path = url_path.toLocalFile()
+            doc_ext = os.path.splitext(doc_path)[1]
+            if doc_ext in get_supported_extensions():
+                documents += [Document(doc_path)]
+
+        # Ignore when all docs are unsupported
+        num_unsupported_docs = len(ev.mimeData().urls()) - len(documents)
+        if len(documents) == 0 or num_unsupported_docs == len(ev.mimeData().urls()):
```

Comment:
Are there circumstances under which both conditions (`len(documents) == 0` and ` num_unsupported_docs == len(ev.mimeData().urls())`) are not both either `True` or `False` (I am struggling to see it from the loop above)? If not, then perhaps it could be simplified?, e.g.:
```suggestion
        if not documents:
```